call-profile/main.go

```diff
@@ -0,0 +1,566 @@
+		})
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+
+		return *results.Parameter.Value
+	} else {
+		return ""
+	}
+}
+
+/*
+ Utils
+*/
+
+/*
+ Function to initialize the firestore client
+*/
+func initializeFirestoreClient(ctx context.Context) (*firestore.Client, error) {
+	sa := option.WithCredentialsJSON([]byte(getFirestoreKey()))
+	app, err := firebase.NewApp(ctx, nil, sa)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := app.Firestore(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (res Res) Validate() error {
+	return validation.ValidateStruct(&res,
+		validation.Field(&res.FirstName, validation.Required),
+		validation.Field(&res.LastName, validation.Required),
+		validation.Field(&res.Phone, validation.Required, is.Digit),
+		validation.Field(&res.Email, validation.Required, is.Email),
+		validation.Field(&res.YOE, validation.Min(0)),
+		validation.Field(&res.Company, validation.Required),
+		validation.Field(&res.Designation, validation.Required),
+		validation.Field(&res.GithubId, validation.Required),
+		validation.Field(&res.LinkedIn, validation.Required),
+		validation.Field(&res.Website, is.URL))
+}
+
+/*
+ MODELS
+*/
+
+/*
+ Logs the health of the user's service
+*/
+func logHealth(client *firestore.Client, ctx context.Context, userId string, isServiceRunning bool, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_SERVICE_HEALTH"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId":    userId,
+			"sessionId": sessionId,
+		},
+		Body: map[string]interface{}{
+			"userId":         userId,
+			"serviceRunning": isServiceRunning,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+/*
+ Logs the status of the user's profileDiff
+*/
+func logProfileSkipped(client *firestore.Client, ctx context.Context, userId string, reason string, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_SKIPPED"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId":    userId,
+			"sessionId": sessionId,
+		},
+		Body: map[string]interface{}{
+			"userId": userId,
+			"reason": reason,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+func logProfileStored(client *firestore.Client, ctx context.Context, userId string, sessionId string) {
+	newLog := Log{
+		Type:      Constants["PROFILE_DIFF_STORED"],
+		Timestamp: time.Now(),
+		Meta: map[string]interface{}{
+			"userId": userId,
+			"sessionId": sessionId
+		},
+		Body: map[string]interface{}{
+			"userId": userId,
+		},
+	}
+	client.Collection("logs").Add(ctx, newLog)
+}
+
+/*
+ Function for setting the profileStatus in user object in firestore
+*/
+func setProfileStatusBlocked(client *firestore.Client, ctx context.Context, userId string, reason string, sessionId string) {
+	client.Collection("users").Doc(userId).Set(ctx, map[string]interface{}{
+		"profileStatus": Constants["STATUS_BLOCKED"],
+		"chaincode":     "",
+	}, firestore.MergeAll)
```

Comment:
MergeAll option will merge the new document with the existing document, but it will not overwrite the existing document. This means that if the profileStatus field already exists in the document, the value of the field will not be changed.

Reply:
Ok, I have to check it. I think I read something wrong.
Thanks.